File: python3/1243.py
while True:
    try:
        frase = input().split()
        qnt = len(frase)
        
        soma = 0
        cont = 0

        for i in range(qnt):

            word = True
            tam = len(frase[i])

            for j in range(tam):
                if(frase[i][j].isalpha() or frase[i][j] == "."):
                    if(frase[i][j].isalpha()):
                        continue
                    else:
                        if(j != tam - 1 or tam == 1):
                            word = False
                            break
                else:
                    word = False
                    break
            
            if(word):
                if(frase[i][tam - 1] == "."):
                    tam -= 1
                soma += tam
                cont += 1

        if(cont != 0):
            media = soma // cont
            if(media <= 3):
                pontos = 250
            elif(media == 4 or media == 5):
                pontos = 500
            elif(media >= 6):
                pontos = 1000
        else:
            pontos = 250        

        print(pontos)  


    except EOFError:
        break

File: c++/1243.cpp
#include <iostream>
#include <cstdio>
#include <cstring>
#include <sstream>

using namespace std;

int main() {

    //freopen("1243.in", "r", stdin);
    string entrada, palavra;

    while(getline(cin, entrada)) {
        stringstream s(entrada);
        int soma = 0;
        int total = 0;

        while(s >> palavra) {
            int tam = palavra.size();

            bool word = true;

            for(int i = 0; i < tam; i++) {
                if(isalpha(palavra[i]) or palavra[i] == '.') {
                    if(isalpha(palavra[i])) {
                        continue;
                    } else {
                        if(i != tam - 1 or tam == 1) {
                            word = false;
                            break;
                        }
                    }
                } else {
                    word = false;
                    break;
                }
            }

            if(word) {
                if(palavra[tam - 1] == '.') {
                    tam--;
                }
                soma += tam;
                total++;
            }
            //cout << palavra << " " << tam << " " << word << endl;
        }
        int pontos;
        if(soma != 0) {
            int media = soma / total;
            if(media <= 3) {
                pontos = 250;
            } else if(media > 3 and media <= 5) {
                pontos = 500;
            } else if(media >= 6) {
                pontos = 1000;
            }
        } else {
            pontos = 250;
        }
        cout << pontos << endl;
    }

    return 0;
}
